src/node_v8.cc

```diff
@@ -295,6 +295,154 @@ static void SetHeapStatistics(JSONWriter* writer, Isolate* isolate) {
+                                                 3);
+
+    // Set Space Statistics
+    Local<v8::Name> spaceStatsNames[] = {
+        FIXED_ONE_BYTE_STRING(isolate, "name"),
+        FIXED_ONE_BYTE_STRING(isolate, "committedSizeBytes"),
+        FIXED_ONE_BYTE_STRING(isolate, "residentSizeBytes"),
+        FIXED_ONE_BYTE_STRING(isolate, "usedSizeBytes"),
+        FIXED_ONE_BYTE_STRING(isolate, "pageStats"),
+        FIXED_ONE_BYTE_STRING(isolate, "freeListStats")};
+    Local<v8::Value> spaceStatsValues[] = {
+        ToV8Value(context, stats.space_stats[i].name.c_str(), isolate)
+            .ToLocalChecked(),
+        ToV8Value(context, stats.space_stats[i].committed_size_bytes, isolate)
+            .ToLocalChecked(),
+        ToV8Value(context, stats.space_stats[i].resident_size_bytes, isolate)
+            .ToLocalChecked(),
+        ToV8Value(context, stats.space_stats[i].used_size_bytes, isolate)
+            .ToLocalChecked(),
+        Array::New(
+            isolate, pageStatisticsArray.data(), pageStatisticsArray.size()),
+        freeListStatsObj,
+    };
+    Local<Object> spaceStatsObject = Object::New(
+        isolate, v8::Null(isolate), spaceStatsNames, spaceStatsValues, 6);
+    spaceStatisticsArray.emplace_back(spaceStatsObject);
+  }
+
+  std::string detailLevel =
+      (stats.detail_level == cppgc::HeapStatistics::DetailLevel::kDetailed)
+          ? "kDetailed"
+          : "kBrief";
+
+  // Set heap statistics
+  Local<v8::Name> heapStatisticsNames[] = {
+      FIXED_ONE_BYTE_STRING(isolate, "committedSizeBytes"),
+      FIXED_ONE_BYTE_STRING(isolate, "residentSizeBytes"),
+      FIXED_ONE_BYTE_STRING(isolate, "usedSizeBytes"),
+      FIXED_ONE_BYTE_STRING(isolate, "detailLevel"),
+      FIXED_ONE_BYTE_STRING(isolate, "spaceStatistics"),
+      FIXED_ONE_BYTE_STRING(isolate, "typeNames")};
+
+  Local<v8::Value> heapStatisticsValues[] = {
+      ToV8Value(context, stats.committed_size_bytes, isolate).ToLocalChecked(),
+      ToV8Value(context, stats.resident_size_bytes, isolate).ToLocalChecked(),
+      ToV8Value(context, stats.used_size_bytes, isolate).ToLocalChecked(),
+      ToV8Value(context, detailLevel, isolate).ToLocalChecked(),
+      Array::New(
+          isolate, spaceStatisticsArray.data(), spaceStatisticsArray.size()),
+      ToV8Value(context, stats.type_names, isolate).ToLocalChecked()};
+
+  Local<Object> heapStatisticsObject = Object::New(
+      isolate, v8::Null(isolate), heapStatisticsNames, heapStatisticsValues, 6);
+
+  return heapStatisticsObject;
+}
+
+static void CollectCppHeapStatistics(
+    const v8::FunctionCallbackInfo<v8::Value>& args) {
+  v8::Isolate* isolate = args.GetIsolate();
+  v8::HandleScope handle_scope(isolate);
+
+  CHECK_EQ(args.Length(), 1);
+  CHECK(args[0]->IsInt32());
+  int detailLevel = args[0].As<v8::Int32>()->Value();
+
+  cppgc::HeapStatistics::DetailLevel detail_level =
+      detailLevel ? cppgc::HeapStatistics::DetailLevel::kDetailed
```

Comment:
If this is only used as a boolean, you could just pass a boolean to C++ land. Though for future extensions it may still be better to expose the DetailLevel enum as constants to JS land. You can check out e.g. how `MeasureMemoryExecution` is handled for `vm.measureMemory` in https://github.com/nodejs/node/blob/269496e8e166fbc06d6f3808a5863ce792de9f11/src/node_contextify.cc#L2013 and https://github.com/nodejs/node/blob/269496e8e166fbc06d6f3808a5863ce792de9f11/src/node_contextify.cc#L1946 for an example.